backend/plugin/db/oracle/oracle.go

```diff
@@ -239,14 +239,15 @@ func (driver *Driver) getOracleStatementWithResultLimit(stmt string, queryContex
 func (driver *Driver) querySingleSQL(ctx context.Context, conn *sql.Conn, singleSQL base.SingleSQL, queryContext *db.QueryContext) (*v1pb.QueryResult, error) {
 	statement := strings.TrimRight(singleSQL.Text, " \n\t;")
 
-	stmt := statement
-	if !strings.HasPrefix(strings.ToUpper(stmt), "EXPLAIN") && queryContext.Limit > 0 {
-		var err error
-		stmt, err = driver.getOracleStatementWithResultLimit(stmt, queryContext)
+	if queryContext.Explain {
+		statement = fmt.Sprintf("EXPLAIN %s", statement)
+	} else if queryContext.Limit > 0 {
+		stmt, err := driver.getOracleStatementWithResultLimit(statement, queryContext)
```

Comment:
Oracle doesn't support EXPLAIN SELECT

Reply:
Correct. This preserves the existing behavior but will be fixed in a follow-up PR.